File: python/problem_03.py
# Largest prime factor
# Problem 3
# The prime factors of 13195 are 5, 7, 13 and 29.
#
# What is the largest prime factor of the number 600851475143 ?

def my_largest_prime(num):
    counter = 2
    sol = []
    while num != 1:
        if num % counter == 0:
            num /= counter
            if counter not in sol:
                sol.append(counter)
            counter = 2
        else:
            counter += 1

    sol.sort()
    return sol[-1]

num = 600851475143
print(my_largest_prime(21)) #  expecting 7
print(my_largest_prime(100)) #  expecting 5
print(my_largest_prime(num)) #  expecting 6857

File: javascript/problem_03.js
// # Largest prime factor
// # Problem 3
// # The prime factors of 13195 are 5, 7, 13 and 29.
// #
// # What is the largest prime factor of the number 600851475143 ?

// var sol = 0
// var num = 600851475143
// var counter = 2
// while (num > 1) {
//   if (num % counter == 0) {
//     sol = counter
//     num /= counter
//   }
//   else {
//     counter += 1
//   }
// }
// console.log(sol)


var my_largest_prime = function(num) {
  var sol = 0
  var counter = 2
  while (num > 1) {
    if (num % counter == 0) {
      sol = counter
      num /= counter
    }
    else {
      counter += 1
    }
  }
  return sol
}

var num = 600851475143
console.log(my_largest_prime(num)) //  expecting 6857
